### DISCUSSION TITLE
Issue with AutoComplete

### DISCUSSION THREAD

**Discussion by phoskee** [zauberzeug/nicegui#1998] [Q&A] [answered]
### Question

I don't know if I should open an issue, when in doubt I ask a question: 
In the NiceGUI site autocomplete works great, in my script the cursor stays in the middle between the entered characters and the autocompleted word. 
Any ideas?
![Issue](https://github.com/zauberzeug/nicegui/assets/125825348/be90acbc-818e-4e6d-b2f5-d3c162de71fe)
```py
select = ui.input(autocomplete=codice).classes(
                'w-full').props('outlined label="Inserisci cognome" disable type="text" mask="XXXXXXXXXXXXXXXX"') #.on('keypress.enter', stampa)
```

**Comment by falkoschindler** [comment, marked as answer]
It took me a while to understand what is going on, but it looks like Quasar is behaving strangely.

But let's start with a minimal reproduction:
```py
ui.input(autocomplete=['ABC']).props('mask=XXX')
```

And here is a reproduction without NiceGUI:
```html
<!DOCTYPE html>
<html>
  <head>
    <link href="https://cdn.jsdelivr.net/npm/quasar@2.14.0/dist/quasar.css" rel="stylesheet" type="text/css" />
  </head>
  <body>
    <div id="q-app">
      <q-input v-model="inputValue" @keydown.tab="perform_autocomplete" mask="XXX" />
    </div>
    <script src="https://cdn.jsdelivr.net/npm/vue@3/dist/vue.global.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/quasar@2.14.0/dist/quasar.umd.js"></script>
    <script>
      const app = Vue.createApp({
        setup() {
          const inputValue = Vue.ref("");
          const perform_autocomplete = (e) => {
            inputValue.value = "ABC";
            e.preventDefault();
          };
          return {
            inputValue,
            perform_autocomplete,
          };
        },
      });
      app.use(Quasar);
      app.mount("#q-app");
    </script>
  </body>
</html>
```

Although we could track this issue in our repo, there's probably not much we can do about it. I tried setting the cursor position explicitly in the `perform_autocomplete` method, but without success. Somehow the mask mechanism seems to intercept.

If this a Quasar bug? I'm not sure. The autocompletion is our implementation after all. But why does the mask interfere with a key handler? So maybe it's still worth reporting it on their repo.

**Comment by phoskee** [comment]
I noticed that they have already worked on a [similar](https://github.com/quasarframework/quasar/issues/15453) problem
